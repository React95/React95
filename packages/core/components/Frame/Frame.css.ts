import { createRainbowSprinkles, defineProperties } from 'rainbow-sprinkles';
import {
  background,
  borderRadius,
  borders,
  colors,
  displayAndBoxModel,
  font,
  outline,
  positioning,
  shadows,
  text,
  zIndices,
} from './props';
import type { SprinklesFn } from 'rainbow-sprinkles/createRuntimeFn';

const displayAndBoxModelAtomicProps = defineProperties({
  dynamicProperties: displayAndBoxModel,
  shorthands: {
    size: ['height', 'width'],
    h: ['height'],
    w: ['width'],
    minH: ['minHeight'],
    minW: ['minWidth'],

    m: ['margin'],
    mr: ['marginRight'],
    ml: ['marginLeft'],
    mt: ['marginTop'],
    mb: ['marginBottom'],
    marginX: ['marginLeft', 'marginRight'],
    marginY: ['marginTop', 'marginBottom'],
    mx: ['marginLeft', 'marginRight'],
    my: ['marginTop', 'marginBottom'],

    p: ['padding'],
    pr: ['paddingRight'],
    pl: ['paddingLeft'],
    pt: ['paddingTop'],
    pb: ['paddingBottom'],
    paddingX: ['paddingLeft', 'paddingRight'],
    paddingY: ['paddingTop', 'paddingBottom'],
    px: ['paddingLeft', 'paddingRight'],
    py: ['paddingTop', 'paddingBottom'],
  },
});

const positioningAtomicProps = defineProperties({
  dynamicProperties: positioning,
});

const colorsAtomicProps = defineProperties({
  dynamicProperties: colors,
  shorthands: {
    bgColor: ['backgroundColor'],
  },
});

const backgroundAtomicProps = defineProperties({
  dynamicProperties: background,
  shorthands: {
    bg: ['background'],
  },
});

const bordersAtomicProps = defineProperties({
  dynamicProperties: borders,
});

const borderRadiusAtomicProps = defineProperties({
  dynamicProperties: borderRadius,
});

const shadowsAtomicProps = defineProperties({
  dynamicProperties: shadows,
});

const outlineAtomicProps = defineProperties({
  dynamicProperties: outline,
});

const fontAtomicProps = defineProperties({
  dynamicProperties: font,
});

const textAtomicProps = defineProperties({
  dynamicProperties: text,
});

const zIndicesAtomicProps = defineProperties({
  dynamicProperties: zIndices,
});

export const sprinkles: SprinklesFn<
  [
    typeof colorsAtomicProps,
    typeof shadowsAtomicProps,
    typeof zIndicesAtomicProps,
    typeof displayAndBoxModelAtomicProps,
    typeof positioningAtomicProps,
    typeof backgroundAtomicProps,
    typeof bordersAtomicProps,
    typeof borderRadiusAtomicProps,
    typeof outlineAtomicProps,
    typeof fontAtomicProps,
    typeof textAtomicProps,
  ]
> = createRainbowSprinkles(
  colorsAtomicProps,
  shadowsAtomicProps,
  zIndicesAtomicProps,
  displayAndBoxModelAtomicProps,
  positioningAtomicProps,
  backgroundAtomicProps,
  bordersAtomicProps,
  borderRadiusAtomicProps,
  outlineAtomicProps,
  fontAtomicProps,
  textAtomicProps,
);

export type Sprinkles = Parameters<typeof sprinkles>[0];
