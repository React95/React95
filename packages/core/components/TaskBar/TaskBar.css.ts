import { globalStyle, style } from '@vanilla-extract/css';
import { recipe } from '@vanilla-extract/recipes';
import { contract } from '../themes/contract.css';
import { calc } from '@vanilla-extract/css-utils';

export const truncate = style({
  overflow: 'hidden',
  display: '-webkit-box',
  WebkitLineClamp: '1',
  WebkitBoxOrient: 'vertical',
  textAlign: 'left',
});

export const tooltip = style({});

globalStyle(`${tooltip} div:first-child`, {
  right: 0,
});

export const windowsButton = recipe({
  base: {
    display: 'inline-flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: contract.colors.material,
    paddingBlock: contract.space[2],
    paddingInline: contract.space[3],
    marginRight: contract.space[2],
    maxWidth: '150px',
    border: 'none',
    outline: 'none',
  },
  variants: {
    small: {
      true: {
        paddingInline: contract.space[5],
      },
      false: {
        width: '100%',
      },
    },
    active: {
      true: {
        boxShadow: contract.shadows.in,
      },
      false: {
        boxShadow: contract.shadows.out,
      },
    },
  },
  compoundVariants: [
    {
      variants: {
        active: true,
        small: true,
      },
      style: {
        paddingTop: contract.space[4],
        paddingBottom: contract.space[0],
        outline: `${contract.space[1]} dotted ${contract.colors.borderDarkest}`,
        outlineOffset: calc.negate(contract.space[4]),
      },
    },
    {
      variants: {
        active: true,
        small: false,
      },
      style: {
        backgroundColor: contract.colors.borderLighter,
      },
    },
  ],
});

export const icon = style({});

globalStyle(`${icon} svg`, {
  marginTop: contract.space[4],
  marginRight: contract.space[4],
  minWidth: contract.space[2],
  width: contract.space[20],
  height: contract.space[20],
  shapeRendering: 'auto',
});
