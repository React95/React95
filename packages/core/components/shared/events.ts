import { ReactElement } from 'react';

export class Emitter<U extends string, T> {
  private listeners: { [key: string]: Array<(data: Partial<T>) => void> } = {};

  on(eventName: U, callback: (data: T) => void) {
    if (!this.listeners[eventName]) {
      this.listeners[eventName] = [];
    }
    this.listeners[eventName].push(callback);
  }

  off(eventName: U, callback: (data: T) => void) {
    if (this.listeners[eventName]) {
      this.listeners[eventName] = this.listeners[eventName].filter(
        cb => cb !== callback,
      );
    }
  }

  emit(eventName: U, data: Partial<T>) {
    if (this.listeners[eventName]) {
      this.listeners[eventName].forEach(callback => callback(data));
    }
  }
}

export enum ModalEvents {
  AddModal = 'add-modal',
  RemoveModal = 'remove-modal',
  ModalVisibilityChanged = 'modal-visibility-changed',
  MinimizeModal = 'minimize-modal',
  RestoreModal = 'restore-modal',
}

export interface ModalWindow {
  icon?: ReactElement;
  title: string;
  hasButton: boolean;
  id: string;
}

export const modals = new Emitter<ModalEvents, ModalWindow>();
