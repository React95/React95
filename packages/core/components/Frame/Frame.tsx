import React, { forwardRef } from 'react';
import type {
  ComponentPropsWithRef,
  ElementType,
  ForwardedRef,
  Ref,
  ReactNode,
  RefAttributes,
} from 'react';
import { Sprinkles, sprinkles } from './Frame.css';
import cn from 'classnames';

type FixedForwardRef = <T, P = object>(
  render: (props: P, ref: Ref<T>) => ReactNode,
) => (props: P & RefAttributes<T>) => ReactNode;

export const fixedForwardRef = forwardRef as FixedForwardRef;

type DistributiveOmit<T, TOmitted extends PropertyKey> = T extends any
  ? Omit<T, TOmitted>
  : never;

export type FrameProps<TAs extends ElementType> = {
  as?: TAs;
} & DistributiveOmit<
  ComponentPropsWithRef<ElementType extends TAs ? 'div' : TAs>,
  'as'
> &
  Sprinkles;

const FrameComponent = <TAs extends ElementType = 'div'>(
  props: FrameProps<TAs>,
  ref: ForwardedRef<any>,
) => {
  const { as, children, ...rest } = props;
  const Component = as || 'div';
  const { className, style, otherProps } = sprinkles(rest);

  return (
    <Component
      {...otherProps}
      style={{ ...style, ...otherProps.style }}
      className={cn(className, otherProps.className)}
      ref={ref}
    >
      {children}
    </Component>
  );
};

export const Frame = fixedForwardRef(FrameComponent);
