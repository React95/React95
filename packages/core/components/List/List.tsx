import React, { forwardRef } from 'react';
import cn from 'classnames';

import { ListItem } from './ListItem';
import { Divider } from './ListDivider';
import { Frame, FrameProps } from '../Frame/Frame';
import { list } from './List.css';

export type ListProps = Omit<FrameProps<'ul'>, 'as'>;

const ListRenderer = forwardRef<HTMLUListElement, ListProps>((rest, ref) => (
  <Frame {...rest} ref={ref} className={cn(list, rest.className)} as="ul" />
));

export type IListProps = typeof ListRenderer & {
  Item: typeof ListItem;
  Divider: typeof Divider;
};

export const List: IListProps = Object.assign(ListRenderer, {
  Item: ListItem,
  Divider: Divider,
});
