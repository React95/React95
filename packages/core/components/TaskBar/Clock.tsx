import React, { useEffect, useState } from 'react';

import { Frame } from '../Frame/Frame';
import { Tooltip } from '../Tooltip/Tooltip';
import { tooltip } from './TaskBar.css';

export const Clock = () => {
  const [timer, setTimer] = useState('');

  useEffect(() => {
    function checkTime(i: number) {
      return i < 10 ? `0${i}` : i;
    }

    const interval = setInterval(() => {
      const today = new Date();
      const h = today.getHours();
      const m = today.getMinutes();
      setTimer(`${checkTime(h)}:${checkTime(m)}`);
    });

    return () => clearInterval(interval);
  }, []);

  return (
    <Frame
      boxShadow="$in"
      px="$6"
      py="$2"
      display="flex"
      justifyContent="center"
      alignItems="center"
    >
      <Tooltip className={tooltip}>{timer}</Tooltip>
    </Frame>
  );
};
