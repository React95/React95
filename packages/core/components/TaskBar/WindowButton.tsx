import React, { forwardRef } from 'react';

import { Windows } from '../Modal/ModalContext';
import * as styles from './TaskBar.css';

export type WindowButtonProps = {
  small?: boolean;
  active?: boolean;
} & Pick<Windows, 'icon'> &
  React.ButtonHTMLAttributes<HTMLButtonElement>;

export const WindowButton = forwardRef<HTMLButtonElement, WindowButtonProps>(
  ({ children, small = false, icon, active = false, ...props }, ref) => (
    <button
      className={styles.windowsButton({ active, small })}
      {...props}
      ref={ref}
    >
      <div className={styles.icon}>{icon}</div>
      {children}
    </button>
  ),
);
