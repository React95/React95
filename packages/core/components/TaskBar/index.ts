import { TaskBar } from './TaskBar';

export default TaskBar;
