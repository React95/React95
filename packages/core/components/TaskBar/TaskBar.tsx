import React, { forwardRef, useEffect, useState } from 'react';
import type { ReactElement } from 'react';

import { Frame } from '../Frame/Frame';
import { List } from '../List/List';

import { Clock } from './Clock';
import { WindowButton } from './WindowButton';
import { Logo } from '@react95/icons';
import { truncate } from './TaskBar.css';
import { ModalEvents, modals, ModalWindow } from '../shared/events';

export type TaskBarProps = {
  list?: ReactElement<typeof List>;
};

export const TaskBar = forwardRef<HTMLDivElement, TaskBarProps>(
  ({ list }, ref) => {
    const [showList, toggleShowList] = useState(false);
    const [activeStart, toggleActiveStart] = useState(false);
    const [modalWindows, setModalWindows] = React.useState<ModalWindow[]>([]);
    const [activeWindow, setActiveWindow] = useState<string>();
    // const [minimizedModals, setMinimizedModals] = useState<ModalWindow[]>([]);

    useEffect(() => {
      const addModal = (window: ModalWindow) => {
        setModalWindows(prevModals => [...prevModals, window]);
      };
      const removeModal = (data: Pick<ModalWindow, 'id'>) => {
        setModalWindows(prevModals => {
          const filteredModals = prevModals.filter(
            modal => modal.id !== data.id,
          );

          const lastModal = filteredModals.at(-1);

          if (!activeWindow && lastModal) {
            modals.emit(ModalEvents.ModalVisibilityChanged, {
              id: lastModal?.id,
            });
          }

          return filteredModals;
        });
      };

      const updateVisibleModal = ({ id }: Pick<ModalWindow, 'id'>) => {
        setActiveWindow(id);
      };

      modals.on(ModalEvents.AddModal, addModal);
      modals.on(ModalEvents.RemoveModal, removeModal);
      modals.on(ModalEvents.ModalVisibilityChanged, updateVisibleModal);

      return () => {
        modals.off(ModalEvents.AddModal, addModal);
        modals.off(ModalEvents.RemoveModal, removeModal);
        modals.off(ModalEvents.ModalVisibilityChanged, updateVisibleModal);
      };
    }, []);

    return (
      <Frame
        position="fixed"
        bottom="0px"
        left="0px"
        right="0px"
        display="flex"
        justifyContent="space-between"
        h="28px"
        w="100%"
        padding="$2"
        zIndex="$taskbar"
        backgroundColor="$material"
        boxShadow="$out"
        ref={ref}
      >
        {showList && (
          <Frame
            position="absolute"
            bottom="28px"
            onClick={() => {
              toggleActiveStart(false);
              toggleShowList(false);
            }}
          >
            {list}
          </Frame>
        )}
        <WindowButton
          small
          icon={<Logo variant="32x32_4" />}
          active={activeStart}
          onClick={() => {
            toggleActiveStart(!activeStart);
            toggleShowList(!showList);
          }}
        >
          Start
        </WindowButton>

        <Frame w="100%" paddingLeft="$0" ml="$2" display="flex">
          {modalWindows.map(
            ({ icon, title, hasButton, id }) =>
              hasButton && (
                <WindowButton
                  key={id}
                  icon={icon}
                  active={id === activeWindow}
                  onClick={() => {
                    if (id === activeWindow) {
                      modals.emit(ModalEvents.MinimizeModal, { id });
                      setActiveWindow('Minimize');
                    } else {
                      modals.emit(ModalEvents.RestoreModal, { id }); 
                      modals.emit(ModalEvents.ModalVisibilityChanged, { id });
                    }
                  }}
                  small={false}
                >
                  <div className={truncate}>{title}</div>
                </WindowButton>
              ),
          )}
        </Frame>

        <Clock />
      </Frame>
    );
  },
);
