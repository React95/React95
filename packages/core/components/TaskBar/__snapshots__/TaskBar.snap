// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TaskBar /> > Snapshots > <Clock /> > should match snapshot 1`] = `
<div>
  <div
    class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
    style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
  >
    <div
      class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
    >
      10:00
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> > Snapshots > <WindowButton /> > should match snapshot 1`] = `
<div>
  <button
    class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_false__1ct83mo4"
  >
    <div
      class="TaskBar_icon__1ct83mo9"
    />
  </button>
</div>
`;

exports[`<TaskBar /> > Snapshots > <WindowButton /> > should match snapshot with active prop 1`] = `
<div>
  <button
    class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
  >
    <div
      class="TaskBar_icon__1ct83mo9"
    />
  </button>
  <button
    class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_true__1ct83mo5 TaskBar_windowsButton_small_true__1ct83mo3 TaskBar_windowsButton_compound_0__1ct83mo7"
  >
    <div
      class="TaskBar_icon__1ct83mo9"
    />
  </button>
</div>
`;

exports[`<TaskBar /> > Snapshots > <WindowButton /> > should match snapshot with icon prop 1`] = `
<div>
  <button
    class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_false__1ct83mo4"
  >
    <div
      class="TaskBar_icon__1ct83mo9"
    >
      <svg test-file-stub />
    </div>
  </button>
</div>
`;

exports[`<TaskBar /> > Snapshots > <WindowButton /> > should match snapshot with small prop 1`] = `
<div>
  <button
    class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
  >
    <div
      class="TaskBar_icon__1ct83mo9"
    />
  </button>
</div>
`;

exports[`<TaskBar /> > Snapshots > should match snapshot 1`] = `
<div>
  <div
    class="Frame_height__iuzg891z Frame_width__iuzg891t Frame_position__iuzg893d Frame_bottom__iuzg893j Frame_left__iuzg893l Frame_right__iuzg893h Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_padding__iuzg892r Frame_zIndex__iuzg897d Frame_backgroundColor__iuzg893p Frame_boxShadow__iuzg895z"
    style="--height__iuzg891y: 28px; --width__iuzg891s: 100%; --position__iuzg893c: fixed; --bottom__iuzg893i: 0px; --left__iuzg893k: 0px; --right__iuzg893g: 0px; --display__iuzg890: flex; --justifyContent__iuzg891e: space-between; --padding__iuzg892q: var(--r95-space-2); --zIndex__iuzg897c: var(--r95-zIndex-taskbar); --backgroundColor__iuzg893o: var(--r95-color-material); --boxShadow__iuzg895y: var(--r95-shadow-out);"
  >
    <button
      class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
    >
      <div
        class="TaskBar_icon__1ct83mo9"
      >
        <svg test-file-stub
          variant="32x32_4"
        />
      </div>
      Start
    </button>
    <div
      class="Frame_width__iuzg891t Frame_marginLeft__iuzg892l Frame_paddingLeft__iuzg8937 Frame_display__iuzg891"
      style="--width__iuzg891s: 100%; --marginLeft__iuzg892k: var(--r95-space-2); --paddingLeft__iuzg8936: var(--r95-space-0); --display__iuzg890: flex;"
    />
    <div
      class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
      style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
    >
      <div
        class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
      >
        10:00
      </div>
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> > Snapshots > should match snapshot with list prop 1`] = `
<div>
  <div
    class="Frame_height__iuzg891z Frame_width__iuzg891t Frame_position__iuzg893d Frame_bottom__iuzg893j Frame_left__iuzg893l Frame_right__iuzg893h Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_padding__iuzg892r Frame_zIndex__iuzg897d Frame_backgroundColor__iuzg893p Frame_boxShadow__iuzg895z"
    style="--height__iuzg891y: 28px; --width__iuzg891s: 100%; --position__iuzg893c: fixed; --bottom__iuzg893i: 0px; --left__iuzg893k: 0px; --right__iuzg893g: 0px; --display__iuzg890: flex; --justifyContent__iuzg891e: space-between; --padding__iuzg892q: var(--r95-space-2); --zIndex__iuzg897c: var(--r95-zIndex-taskbar); --backgroundColor__iuzg893o: var(--r95-color-material); --boxShadow__iuzg895y: var(--r95-shadow-out);"
  >
    <button
      class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
    >
      <div
        class="TaskBar_icon__1ct83mo9"
      >
        <svg test-file-stub
          variant="32x32_4"
        />
      </div>
      Start
    </button>
    <div
      class="Frame_width__iuzg891t Frame_marginLeft__iuzg892l Frame_paddingLeft__iuzg8937 Frame_display__iuzg891"
      style="--width__iuzg891s: 100%; --marginLeft__iuzg892k: var(--r95-space-2); --paddingLeft__iuzg8936: var(--r95-space-0); --display__iuzg890: flex;"
    />
    <div
      class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
      style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
    >
      <div
        class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
      >
        10:00
      </div>
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> > Snapshots > should match snapshot with list prop 2`] = `
<div>
  <div
    class="Frame_height__iuzg891z Frame_width__iuzg891t Frame_position__iuzg893d Frame_bottom__iuzg893j Frame_left__iuzg893l Frame_right__iuzg893h Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_padding__iuzg892r Frame_zIndex__iuzg897d Frame_backgroundColor__iuzg893p Frame_boxShadow__iuzg895z"
    style="--height__iuzg891y: 28px; --width__iuzg891s: 100%; --position__iuzg893c: fixed; --bottom__iuzg893i: 0px; --left__iuzg893k: 0px; --right__iuzg893g: 0px; --display__iuzg890: flex; --justifyContent__iuzg891e: space-between; --padding__iuzg892q: var(--r95-space-2); --zIndex__iuzg897c: var(--r95-zIndex-taskbar); --backgroundColor__iuzg893o: var(--r95-color-material); --boxShadow__iuzg895y: var(--r95-shadow-out);"
  >
    <div
      class="Frame_position__iuzg893d Frame_bottom__iuzg893j"
      style="--position__iuzg893c: absolute; --bottom__iuzg893i: 28px;"
    >
      <ul
        class="List_list__1lxkvz42"
      >
        <li
          class="List_listItem__1lxkvz40"
        >
          <svg test-file-stub />
          Local Disk (C:)
        </li>
        <li
          class="List_listItem__1lxkvz40"
        >
          <svg test-file-stub />
          Windows Explorer
        </li>
      </ul>
    </div>
    <button
      class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_true__1ct83mo5 TaskBar_windowsButton_small_true__1ct83mo3 TaskBar_windowsButton_compound_0__1ct83mo7"
    >
      <div
        class="TaskBar_icon__1ct83mo9"
      >
        <svg test-file-stub
          variant="32x32_4"
        />
      </div>
      Start
    </button>
    <div
      class="Frame_width__iuzg891t Frame_marginLeft__iuzg892l Frame_paddingLeft__iuzg8937 Frame_display__iuzg891"
      style="--width__iuzg891s: 100%; --marginLeft__iuzg892k: var(--r95-space-2); --paddingLeft__iuzg8936: var(--r95-space-0); --display__iuzg890: flex;"
    />
    <div
      class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
      style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
    >
      <div
        class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
      >
        10:00
      </div>
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> > Snapshots > should match snapshot with one Modal 1`] = `
<div>
  <div
    class="Modal_modalWrapper__1txblt60 Modal_modalWrapper_active_true__1txblt61 Modal_modalWrapper_minimized_false__1txblt63 neodrag"
    style="transform: translate3d(0px, 0px, 0); touch-action: none;"
  >
    <div
      class="TitleBar_titleBarBackground__1kpzw682 TitleBar_titleBarBackground_active_true__1kpzw683 draggable"
    >
      <svg test-file-stub />
      <div
        class="TitleBar_title__1kpzw685"
      >
        file.bat
      </div>
    </div>
  </div>
  <div
    class="Frame_height__iuzg891z Frame_width__iuzg891t Frame_position__iuzg893d Frame_bottom__iuzg893j Frame_left__iuzg893l Frame_right__iuzg893h Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_padding__iuzg892r Frame_zIndex__iuzg897d Frame_backgroundColor__iuzg893p Frame_boxShadow__iuzg895z"
    style="--height__iuzg891y: 28px; --width__iuzg891s: 100%; --position__iuzg893c: fixed; --bottom__iuzg893i: 0px; --left__iuzg893k: 0px; --right__iuzg893g: 0px; --display__iuzg890: flex; --justifyContent__iuzg891e: space-between; --padding__iuzg892q: var(--r95-space-2); --zIndex__iuzg897c: var(--r95-zIndex-taskbar); --backgroundColor__iuzg893o: var(--r95-color-material); --boxShadow__iuzg895y: var(--r95-shadow-out);"
  >
    <button
      class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
    >
      <div
        class="TaskBar_icon__1ct83mo9"
      >
        <svg test-file-stub
          variant="32x32_4"
        />
      </div>
      Start
    </button>
    <div
      class="Frame_width__iuzg891t Frame_marginLeft__iuzg892l Frame_paddingLeft__iuzg8937 Frame_display__iuzg891"
      style="--width__iuzg891s: 100%; --marginLeft__iuzg892k: var(--r95-space-2); --paddingLeft__iuzg8936: var(--r95-space-0); --display__iuzg890: flex;"
    />
    <div
      class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
      style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
    >
      <div
        class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
      >
        10:00
      </div>
    </div>
  </div>
</div>
`;

exports[`<TaskBar /> > Snapshots > should match snapshot with two Modals 1`] = `
<div>
  <div
    class="Modal_modalWrapper__1txblt60 Modal_modalWrapper_minimized_false__1txblt63 neodrag"
    style="transform: translate3d(0px, 0px, 0); touch-action: none;"
  >
    <div
      class="TitleBar_titleBarBackground__1kpzw682 TitleBar_titleBarBackground_active_false__1kpzw684 draggable"
    >
      <svg test-file-stub />
      <div
        class="TitleBar_title__1kpzw685"
      >
        Windows Explorer
      </div>
    </div>
  </div>
  <div
    class="Modal_modalWrapper__1txblt60 Modal_modalWrapper_active_true__1txblt61 Modal_modalWrapper_minimized_false__1txblt63 neodrag"
    style="transform: translate3d(0px, 0px, 0); touch-action: none;"
  >
    <div
      class="TitleBar_titleBarBackground__1kpzw682 TitleBar_titleBarBackground_active_true__1kpzw683 draggable"
    >
      <svg test-file-stub />
      <div
        class="TitleBar_title__1kpzw685"
      >
        file.bat
      </div>
    </div>
  </div>
  <div
    class="Frame_height__iuzg891z Frame_width__iuzg891t Frame_position__iuzg893d Frame_bottom__iuzg893j Frame_left__iuzg893l Frame_right__iuzg893h Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_padding__iuzg892r Frame_zIndex__iuzg897d Frame_backgroundColor__iuzg893p Frame_boxShadow__iuzg895z"
    style="--height__iuzg891y: 28px; --width__iuzg891s: 100%; --position__iuzg893c: fixed; --bottom__iuzg893i: 0px; --left__iuzg893k: 0px; --right__iuzg893g: 0px; --display__iuzg890: flex; --justifyContent__iuzg891e: space-between; --padding__iuzg892q: var(--r95-space-2); --zIndex__iuzg897c: var(--r95-zIndex-taskbar); --backgroundColor__iuzg893o: var(--r95-color-material); --boxShadow__iuzg895y: var(--r95-shadow-out);"
  >
    <button
      class="TaskBar_windowsButton__1ct83mo2 TaskBar_windowsButton_active_false__1ct83mo6 TaskBar_windowsButton_small_true__1ct83mo3"
    >
      <div
        class="TaskBar_icon__1ct83mo9"
      >
        <svg test-file-stub
          variant="32x32_4"
        />
      </div>
      Start
    </button>
    <div
      class="Frame_width__iuzg891t Frame_marginLeft__iuzg892l Frame_paddingLeft__iuzg8937 Frame_display__iuzg891"
      style="--width__iuzg891s: 100%; --marginLeft__iuzg892k: var(--r95-space-2); --paddingLeft__iuzg8936: var(--r95-space-0); --display__iuzg890: flex;"
    />
    <div
      class="Frame_paddingLeft__iuzg8937 Frame_paddingRight__iuzg8939 Frame_paddingTop__iuzg893b Frame_paddingBottom__iuzg892z Frame_boxShadow__iuzg895z Frame_display__iuzg891 Frame_justifyContent__iuzg891f Frame_alignItems__iuzg891n"
      style="--paddingLeft__iuzg8936: var(--r95-space-6); --paddingRight__iuzg8938: var(--r95-space-6); --paddingTop__iuzg893a: var(--r95-space-2); --paddingBottom__iuzg892y: var(--r95-space-2); --boxShadow__iuzg895y: var(--r95-shadow-in); --display__iuzg890: flex; --justifyContent__iuzg891e: center; --alignItems__iuzg891m: center;"
    >
      <div
        class="Tooltip_wrapper__f5whxn1 TaskBar_tooltip__1ct83mo1"
      >
        10:00
      </div>
    </div>
  </div>
</div>
`;
